internal/resources/kustomization/resource_kustomization.go

```diff
@@ -40,3 +55,419 @@ var kustomizationSchema = map[string]*schema.Schema{
 	spec.SpecKey:         spec.SpecSchema,
 	statusKey:            status.StatusSchema,
 }
+
+func resourceKustomizationRead(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	kustomizationName, ok := d.Get(nameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read kustomization name")
+	}
+
+	kustomizationNamespaceName, ok := d.Get(namespaceNameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read kustomization namespace name")
+	}
+
+	scopedFullnameData := scope.ConstructScope(d, kustomizationName, kustomizationNamespaceName)
+
+	if scopedFullnameData == nil {
+		return diag.Errorf("Unable to get Tanzu Mission Control kustomization entry; Scope full name is empty")
+	}
+
+	err := enableContinuousDelivery(&config, scopedFullnameData, common.ConstructMeta(d))
```

Comment:
in the case of data source read, we should not enable the feature if not enabled

Reply:
Agreed.

Reply:
Updated.